1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,224 @@
+
+  //endregion Property "path"
+
+  //region Property "size"
+
+  /**
+   * Slot for the {@code size} property.
+   * @see #getSize
+   * @see #setSize
+   */
+  public static final Property size = newProperty(0, 0, null);
+
+  /**
+   * Get the {@code size} property.
+   * @see #size
+   */
+  public int getSize() { return getInt(size); }
+
+  /**
+   * Set the {@code size} property.
+   * @see #size
+   */
+  public void setSize(int v) { setInt(size, v, null); }
+
+  //endregion Property "size"
+
+  //region Property "createdDate"
+
+  /**
+   * Slot for the {@code createdDate} property.
+   * @see #getCreatedDate
+   * @see #setCreatedDate
+   */
+  public static final Property createdDate = newProperty(0, BAbsTime.NULL, null);
+
+  /**
+   * Get the {@code createdDate} property.
+   * @see #createdDate
+   */
+  public BAbsTime getCreatedDate() { return (BAbsTime)get(createdDate); }
+
+  /**
+   * Set the {@code createdDate} property.
+   * @see #createdDate
+   */
+  public void setCreatedDate(BAbsTime v) { set(createdDate, v, null); }
+
+  //endregion Property "createdDate"
+
+  //region Property "modifiedDate"
+
+  /**
+   * Slot for the {@code modifiedDate} property.
+   * @see #getModifiedDate
+   * @see #setModifiedDate
+   */
+  public static final Property modifiedDate = newProperty(0, BAbsTime.NULL, null);
+
+  /**
+   * Get the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public BAbsTime getModifiedDate() { return (BAbsTime)get(modifiedDate); }
+
+  /**
+   * Set the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public void setModifiedDate(BAbsTime v) { set(modifiedDate, v, null); }
+
+  //endregion Property "modifiedDate"
+
+  //region Action "touch"
+
+  /**
+   * Slot for the {@code touch} action.
+   * @see #touch()
+   */
+  public static final Action touch = newAction(0, null);
+
+  /**
+   * Invoke the {@code touch} action.
+   * @see #touch
+   */
+  public void touch() { invoke(touch, null, null); }
+
+  //endregion Action "touch"
+
+  //region Action "print"
+
+  /**
+   * Slot for the {@code print} action.
+   * @see #print()
+   */
+  public static final Action print = newAction(0, null);
+
+  /**
+   * Invoke the {@code print} action.
+   * @see #print
+   */
+  public void print() { invoke(print, null, null); }
+
+  //endregion Action "print"
+
+  //region Type
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BFile.class);
+
+  //endregion Type
+
+//@formatter:on
+//endregion /*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+    public void doTouch() {
+
+        BAbsTime now = BAbsTime.now();
+        if (getCreatedDate() == BAbsTime.NULL) {
+            setCreatedDate(now);
+        }
+
+        setModifiedDate(now);
+    }
+
+    public void doPrint() {
+
+        System.out.println("File Summary");
+
+        for (Slot slot : getSlots()) {
+          if(slot.isProperty())
```

Comment:
You could save a line by iterating over only properties instead of iterating over all slots and then checking if each is a property:

```java
for (Property property : getProperties()) {
  // ...
}
```